Book Store: building a book library app

SofiBretz opened the pull request:
## Book Store: Building a book library app

In this project we:
- Created a table that contains and updates the books that the user adds to the library app.
- Created a buttons to mark as read or unread. 
- Created a Delete button, so the user can delete a book.
- Created an Alert, this alert asks the user if they are sure about deleting the book information.
- Added a Local Storage on the app.
- Created a readme file.
- Rubocop fixed, no offenses detected.

Ready to code review :) Thanks! 

Regards,

Sofi and Andres.

bolah2009 commented on js/library.js:
```diff
@@ -0,0 +1,76 @@
+  let buttons= document.querySelectorAll(".readButton")
+  buttons.forEach(buttonElement => buttonElement.addEventListener("click", updateReadStatus));
+  let deletebuttons= document.querySelectorAll(".deleteButton")
+  deletebuttons.forEach(deleteElement => deleteElement.addEventListener("click", deleteBook));
```
The render function seems to be doing so many things... :thinking: 

We could have functions that create a single book, add event listeners and call them in the render function.

bolah2009 commented on .eslintrc.json:
```diff
@@ -0,0 +1,17 @@
+      "no-param-reassign": "off",
+      "eol-last": "off"
+    }
+  }
```
Please let's use the [**correct config files**](https://github.com/microverseinc/linters-config/tree/master/javascript).

> Do not make any changes in config files - they represent style guidelines that you share with your team - which is a group of all Microverse students. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/general/linter_config.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/54883773/))

bolah2009 commented:
Hi 🙋🏽‍♂

Could you investigate `--fix` flag? See https://eslint.org/docs/user-guide/command-line-interface#fixing-problems

I see you have a lot of linter errors, using the `fix` flag will help with most formatting errors you are having and more.

```sh
# If eslint is setup properly, the following command should help with most of the errors you having.
npx eslint . --fix
```


___

#### This is not a CRR

SofiBretz commented:
### Summary of changes:

- Added a function for the cells and rows and put it out of the render function, so the render function is cleaner.
- Added Validations to the form, so the user can't add empty information required in the form.
- Readme updated.
- We turn "On" the Stickler on this project, apologies for that, we were secure that we tun in On. 
- Linters' problems fixed, no offenses detected.

Ready for a code review, thank you!

Regards,

Sofi and Andres.

bellom commented on js/library.js:
```diff
@@ -0,0 +1,95 @@
+function deleteBook(deletebook) {
+  const button = deletebook.target;
+  const index = button.getAttribute('data-index');
+  alert('Are you sure you want to delete this book?');
```
Let remove `alert` it is not a best practice to have 'alert' in production, part of the reason you decided to edit the config linter file which you're not supposed to change. @bolah2009 mentioned this too.
https://github.com/andynarf/library/pull/1#discussion_r372239324

bolah2009 replied:
Thanks @bellom, I didn't mention this because the linter will but seems it has been disabled on the config.

bellom commented on .eslintrc.json:
```diff
@@ -0,0 +1,23 @@
+    "extends": ["airbnb-base"],
+    "rules": {
+      "no-shadow": "off",
+      "no-param-reassign": "off",
```
https://github.com/andynarf/library/pull/1#discussion_r372239324
Please let us make use of the Microverse config files 👍.

andynarf commented:
##Summary of changes:

- fixed most lint errors.
- added to rules on the lint file the ones that had problems. (shown on image)
                - (both functions call each other, so no matter which one goes first, we would 
                 always get the call before defined error)
                - ( show error doesn't need to be defined by us , as shown on this answer: 
                   https://stackoverflow.com/questions/27965834/show-error-using-javascript)
- deleted all alerts.
- fixed some functions that weren't called.

Ready for a code review, thank you!

Regards,

Sofi and Andres.

<img width="916" alt="image" src="https://user-images.githubusercontent.com/51463886/73473740-081f0100-436c-11ea-8c9d-269148b73c15.png">
